### PR TITLE
SUBMARINE-174. Add travis support for K8s submitter

### PR DESCRIPTION
**Pull request by jiwq**
### What is this PR for?
K8s submitter is ready for POC, but cannot run the UTs in travis. 

### What type of PR is it?
[ Improvement ]

### Todos
https://issues.apache.org/jira/browse/SUBMARINE-300

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-174

### How should this be tested?
https://travis-ci.org/jiwq/submarine/builds/616270819

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: submarine-server/server-submitter/submitter-k8s/src/main/java/org/apache/submarine/server/submitter/k8s/K8sJobSubmitter.java [hunk 1]
```diff
@@ -116,4 +127,43 @@ CustomResourceJobList listCustomResourceJobs(K8sJobRequest request) {
     }
     return null;
   }
+
+  @VisibleForTesting
+  void createCustomResourceDefinition(String filePath) {
```
Comment (yuanzac) [inline submarine-server/server-submitter/submitter-k8s/src/main/java/org/apache/submarine/server/submitter/k8s/K8sJobSubmitter.java]: Is this method added for the future use? We'd better add UT for this method like others

**Comment by liuxunorg** [comment]
@jiwq Thank you contribute this feaute.

Now travis contains a lot of different test jobs, but they all use the same test environment, for example, mysql is installed in all travis job environment. This will cause travis to run longer.
I plan to split the environment of each test job. For example, only the test environment of the submarine server needs to install mysql.

Therefore, please split the submitter-k8s test with a separate job. Only this travis job needs to install k8s.

The other jobs in travis are split, and an issue is created.

File: submarine-server/server-submitter/submitter-k8s/src/main/java/org/apache/submarine/server/submitter/k8s/K8sJobSubmitter.java (hunk 1 above)
Comment (jiwq) [reply submarine-server/server-submitter/submitter-k8s/src/main/java/org/apache/submarine/server/submitter/k8s/K8sJobSubmitter.java]: Nowadays it's unused, so  I'll remove these methods.

**Comment by jiwq** [comment]
Thanks for @yuanzac and @liuxunorg review this PR. I had fixed all points.

**Comment by liuxunorg** [comment]
https://api.travis-ci.org/v3/job/616278636/log.txt
have two test ApiException. Why did you create crd, but `kubernetes.client` can't get it?

```
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running org.apache.submarine.server.submitter.k8s.K8SJobSubmitterTest
[ERROR] 2019-11-24 14:40:04,715 method:org.apache.submarine.server.submitter.k8s.K8sJobSubmitter.getCustomResourceJob(K8sJobSubmitter.java:83)
Get CRD job: Not Found
io.kubernetes.client.ApiException: Not Found
	at io.kubernetes.client.ApiClient.handleResponse(ApiClient.java:882)
	at io.kubernetes.client.ApiClient.execute(ApiClient.java:798)
	at io.kubernetes.client.apis.CustomObjectsApi.getNamespacedCustomObjectWithHttpInfo(CustomObjectsApi.java:1366)
	at io.kubernetes.client.apis.CustomObjectsApi.getNamespacedCustomObject(CustomObjectsApi.java:1348)
	at org.apache.submarine.server.submitter.k8s.K8sJobSubmitter.getCustomResourceJob(K8sJobSubmitter.java:78)
	at org.apache.submarine.server.submitter.k8s.K8SJobSubmitterTest.getCustomJob(K8SJobSubmitterTest.java:98)
	at org.apache.submarine.server.submitter.k8s.K8SJobSubmitterTest.testDeleteCustomJob(K8SJobSubmitterTest.java:86)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:283)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:173)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:128)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:203)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:155)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
[ERROR] 2019-11-24 14:40:04,907 method:org.apache.submarine.server.submitter.k8s.K8sJobSubmitter.getCustomResourceJob(K8sJobSubmitter.java:83)
Get CRD job: Not Found
io.kubernetes.client.ApiException: Not Found
	at io.kubernetes.client.ApiClient.handleResponse(ApiClient.java:882)
	at io.kubernetes.client.ApiClient.execute(ApiClient.java:798)
	at io.kubernetes.client.apis.CustomObjectsApi.getNamespacedCustomObjectWithHttpInfo(CustomObjectsApi.java:1366)
	at io.kubernetes.client.apis.CustomObjectsApi.getNamespacedCustomObject(CustomObjectsApi.java:1348)
	at org.apache.submarine.server.submitter.k8s.K8sJobSubmitter.getCustomResourceJob(K8sJobSubmitter.java:78)
	at org.apache.submarine.server.submitter.k8s.K8SJobSubmitterTest.getCustomJob(K8SJobSubmitterTest.java:98)
	at org.apache.submarine.server.submitter.k8s.K8SJobSubmitterTest.testCreateCustomJob(K8SJobSubmitterTest.java:58)
	at org.apache.submarine.server.submitter.k8s.K8SJobSubmitterTest.testGetCustomJob(K8SJobSubmitterTest.java:70)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:283)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:173)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:128)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:203)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:155)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.858 sec - in org.apache.submarine.server.submitter.k8s.K8SJobSubmitterTest
```

**Comment by jiwq** [comment]
> Why did you create crd, but kubernetes.client can't get it?

@liuxunorg It try to get the CRD object (not the CRD itself) from cluster. The exception caused by the cluster doesn't have the CRD object. Before the two UTs, `testCreateCustomJob` and `testDeleteCustomJob`, executed we should ensure the CRD job whether exists in cluster. So we can ignore this.

**Comment by liuxunorg** [comment]
I think that if we now add support for kubeflow, we need to add a test for kubeflow in the submarine.
Can you create a new jira for kubeflow test?

**Comment by jiwq** [comment]
> I think that if we now add support for kubeflow, we need to add a test for kubeflow in the submarine.
> Can you create a new jira for kubeflow test?

Good idea, I had created the JIRA to tracking this. https://issues.apache.org/jira/browse/SUBMARINE-300

**Pull request closed without merging**